mlos_bench/mlos_bench/launcher.py

```diff
@@ -337,17 +344,13 @@ def _load_optimizer(self, args_optimizer: Optional[str]) -> Optimizer:
         in the --optimizer command line option. If config file not specified,
         create a one-shot optimizer to run a single benchmark trial.
         """
-        if 'max_iterations' in self.global_config:
-            self.global_config['max_iterations'] *= self.trial_config_repeat_count
```

Comment:
Discussed offline:

- previously (#642) we artificially increased the iteration count so that when we repeated trials it wouldn't lessen the number of configs we sampled
- now that we're separating the optimizer (suggest config) and scheduler (repeat of configs), the we can move the tracking of number of configs suggested to the `suggest()` routine in the optimizer and `number of trials executed` to the scheduler
- that opens up the door for more flexible stopping conditions (#427)
- to do that, we should also cleanup our terminology and rename `max_iterations` to `num_config_suggestions` and add a new `max_trials` argument (separate PR though I think)